internal/host/host.go

```diff
@@ -555,16 +555,15 @@ func (m *Manager) SetUploadLogsAt(ctx context.Context, h *models.Host, db *gorm.
 
 func (m *Manager) UpdateConnectivityReport(ctx context.Context, h *models.Host, connectivityReport string) error {
 	if h.Connectivity != connectivityReport {
-		//retrieve cluster
-		var cluster common.Cluster
-		err := m.db.First(&cluster, "id = ?", h.ClusterID).Error
-		if err != nil {
-			return errors.Wrapf(err, "failed to find cluster %s", h.ClusterID)
-		}
 		if len(connectivityReport) > 0 {
+			//retrieve cluster
+			cluster, err := common.GetClusterFromDB(m.db, h.ClusterID, common.UseEagerLoading)
+			if err != nil {
+				m.log.WithError(err).Errorf("unable to process connectivity report: failed to find cluster %s", h.ClusterID)
+				return common.NewApiError(http.StatusNotFound, err)
```

Comment:
```suggestion
				return err
```
It can be 500 as well, i think that the upper level should handle the error.

Reply:
changed as suggested.